Blogs page

mohammad-jundi opened the pull request:
Connected blogs to WP API 
Style added 
for the Search bar function, I will do it in a separate PR

mohammad-jundi committed:
added blogs-page component, fetched data from WP

mohammad-jundi committed:
fixed fetching problem

mohammad-jundi committed:
added style to blogs

mohammad-jundi committed:
Merge branch 'dev' into blogs-page

mohammad-jundi committed:
fixed some bugs

mohammad-jundi committed:
naming improved, bugs fixed

mohammad-jundi committed:
updated with dev

mohammad-jundi committed:
fixed some bugs

mohammad-jundi committed:
yarn fix and test done

mohammad-jundi commented:
![image](https://user-images.githubusercontent.com/62549249/98738184-61669700-23b8-11eb-9d1e-40ceba68a4e8.png)

mohammad-jundi committed:
Image name changed in Blogs

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+  useEffect(() => {
+    fetchBlogs();
+  }, []);
+  const NUM_BLOGS_TO_SHOW = 9;
```
consts with uppercase should be outside of function just below the import

mohammad-jundi replied:
done

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+    const data = await fetch(URL);
+    const fetchedBlogs = await data.json();
+    setBlogs(fetchedBlogs);
+    console.log(fetchedBlogs);
```
please delete console.log here and on line 21

mohammad-jundi replied:
fixed

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+
+export default function Blogs() {
+  const [blogs, setBlogs] = useState([]);
+  const URL =
```
maybe smth like BLOGS_URL could be more descriptive?

louisrli replied:
these constants dont go in fn body, move under imports after line 4

mohammad-jundi replied:
fixed

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+    <div className="blogsMainContainer">
+      <h2 className="ourBlogTitle">Our Blog</h2>
+      <p>
+        Infomration about violence, women and health. Read different topics
```
typo in information. also, you should add translation

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+      <p>
+        Infomration about violence, women and health. Read different topics
+        <br />
+        that contribute positivley to the mental health of women.
```
typo in positively

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+              <h3 className="blogTitles">{blog.title.rendered}</h3>
+              <img
+                className="blogImgs"
+                alt="blog Img"
```
Blog image

mohammad-jundi replied:
this name caused a problem when running the tests so I had to change it to this name since the tag is already for images

jennetturkkan replied:
in alt? i don't think that `alt="Blog image"` will create some problem

mohammad-jundi replied:
![image](https://user-images.githubusercontent.com/62549249/98863502-866f0e80-2479-11eb-84ae-bc2ed4e4ab2c.png)

mohammad-jundi replied:
after changing the name in alt, the test passes

jennetturkkan replied:
you can use blog.title.rendered as an alt. it won't cause error and will be unique for every blog post.

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+              <div className="shadowEffect"></div>
+              <h3 className="blogTitles">{blog.title.rendered}</h3>
+              <img
+                className="blogImgs"
```
i would say this should be blogImg since you map and create blog one by one. same with blogTitle

mohammad-jundi replied:
it makes much more sense, thanks for the suggestion

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+                className="blogImgs"
+                alt="blog Img"
+                src={blog.jetpack_featured_media_url}
+              ></img>
```
img is a self-closing tag

jennetturkkan commented on src/components/Blogs/index.scss:
```diff
@@ -0,0 +1,56 @@
+    flex-wrap: wrap;
+  }
+
+  .col {
```
i didn't see this col in jsx file. where does it come from?

mohammad-jundi replied:
these from bootstrap, I will comments if that's allowed, will ask Ammar about it.

jennetturkkan replied:
but you are not using it anywhere. it's not even imported in index.jsx file

mohammad-jundi replied:
it comes with the package I guess, I am not sure. however when I inspected it and tried to style it worked.

jennetturkkan commented on src/components/Blogs/index.scss:
```diff
@@ -0,0 +1,56 @@
+
+  .blogTitles {
+    text-align: center;
+    position: absolute;
```
i wonder if this position absolute will create any issues in future? i had to delete it from seekHelp section in homepage because it was rendering not the way we wanted

louisrli replied:
absolute position should generally be avoided where possible because it makes code hard to change

mohammad-jundi replied:
I used it inside nesting to avoid any kind of conflict in the future. tried different things, absolute was the most accurate one to use. I'll search if there is something else I can use

jennetturkkan replied:
my concern is that it will create a lot of problems with responsiveness. but it's your call in any case.

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+        {fetchedBlogsArr.map((blog) => {
+          return (
+            <div className="blogsContainer">
+              <div className="shadowEffect"></div>
```
is there other way of adding shadow effect? it looks like this div can be removed here

mohammad-jundi replied:
tried to use box-shadow, didn't work for me. this one worked

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+      <Container>
+        {fetchedBlogsArr.map((blog) => {
+          return (
+            <div className="blogsContainer">
```
I think class name is a bit confusing since you have already bootstrap container above this div

mohammad-jundi replied:
it's a container that contains the blogs container, if you have any suggestions please let me know them, I can change it to blogsSection if it's much more convenient.

jennetturkkan replied:
yeah, it's more clear. thank you

jennetturkkan commented on src/components/Blogs/index.scss:
```diff
@@ -0,0 +1,56 @@
+    margin-left: 1.5%;
+  }
+
+  .mr-sm-2 {
```
if this is a bootstrap class name, you should add comment above it `// Bootstrap class`

mohammad-jundi replied:
done

louisrli commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+import "./index.scss";
+
+export default function Blogs() {
+  const [blogs, setBlogs] = useState([]);
```
so imagine you are me. what is blogs? all i know is blogs is an empty array. add a comment please abovei t

also we usually use "blogPosts" to refer to the posts and blog to refer to the whole site specifically.

mohammad-jundi replied:
fixed

louisrli reviewed (commented):
+jennets comments

mohammad-jundi committed:
fixed bugs

mohammad-jundi committed:
snapshot updated

mohammad-jundi committed:
naming improved

mohammad-jundi committed:
changed the naming for img, h3, added translate

mohammad-jundi commented:
PTAL @louisrli

louisrli commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+import { Container } from "react-bootstrap";
+import { useTranslation } from "react-i18next";
+import "./index.scss";
+const NUM_BLOGS_TO_SHOW = 9;
```
newline

louisrli commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+import { useTranslation } from "react-i18next";
+import "./index.scss";
+const NUM_BLOGS_TO_SHOW = 9;
+const BLOGS_URL =
```
BLOG_API_URL

louisrli commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+export default function Blogs() {
+  const { t } = useTranslation();
+  const [blogPosts, setBlogPosts] = useState([]);
+  const fetchBlogPosts = async () => {
```
consider a newline above here too. it's like a "paragraph" in an essay. you're doing something new here so you separate it from the thing above.

louisrli commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+  useEffect(() => {
+    fetchBlogPosts();
+  }, []);
+  const fetchedBlogPostsArr = blogPosts.slice(0, NUM_BLOGS_TO_SHOW);
```
this should be shownBlogPostsArr. blogPosts is actually the fetchedBlogPosts, but you are only showing N of them.

but actually, you shouldn't do it like this. you should use `per_page` in the API. even without knowing the wordpress API it's possible to know that there's probably something like this because usually APIs dont make you load EVERYTHING then cut it in your code, they make you say exactly how much you want to load beforehand (this saves them network)

see per_page

https://developer.wordpress.org/rest-api/reference/posts/#arguments

mohammad-jundi replied:
I read this documentation before start with WP API and wasn't so clear to me. 
Thanks for referring it to me again it helped me a lot in cleaning the code. Maybe it's not the cleanest but I am sure it's much better now

louisrli commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+    <div className="blogsMainContainer">
+      <h2 className="ourBlogTitle">Our Blog</h2>
+      <p>
+        Information about violence, women and health. Read different topics
```
translations?

louisrli commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,48 @@
+      <h2 className="ourBlogTitle">Our Blog</h2>
+      <p>
+        Information about violence, women and health. Read different topics
+        <br />
```
don't manually line break like this. let the CSS take care of it. it also doesn't play well with translation

mohammad-jundi committed:
naming imroved, fixed styling, fixed bugs

mohammad-jundi committed:
test updated, fixed bugs

mohammad-jundi commented:
PTAL @louisrli 
fixed the extra `&nbsp;` that was showing in titles. 
I am not sure if I did the translation part in the right way, 
I will add the search function in Containers/BlogPage, please advise if it's fine in this way 
![image](https://user-images.githubusercontent.com/62549249/99001834-9277d080-254c-11eb-8ef6-f26d62ad8021.png)

mohammad-jundi committed:
Merge branch 'dev' into blogs-page

jennetturkkan commented on src/containers/BlogPage/index.jsx:
```diff
@@ -0,0 +1,21 @@
+    <div className="blogsMainContainer">
+      <h2 className="ourBlogTitle">Our Blog</h2>
+      <p className="ourBlogSubtitle">
+        {t(`Information about violence, women and health. Read different topics
```
this is not right. you can check how we did translation before. you need to add this text to translation.json and then select it here.

mohammad-jundi replied:
I wanted to do it in this way, but I wasn't sure if this will cause a conflict in the JSON file since Hiba was adding new data. So I kept it in this way till she finishes adding the data so that I can add to it later.

louisrli replied:
just write to en.json first.

jennetturkkan commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,43 @@
+  };
+  useEffect(() => {
+    fetchBlogPosts();
+  }, []);
```
add new line after this

jennetturkkan commented on src/containers/BlogPage/index.jsx:
```diff
@@ -0,0 +1,21 @@
+
+  return (
+    <div className="blogsMainContainer">
+      <h2 className="ourBlogTitle">Our Blog</h2>
```
you need to add translation here too

jennetturkkan commented on src/containers/BlogPage/index.scss:
```diff
@@ -0,0 +1,21 @@
+  }
+
+  .ourBlogSubtitle {
+    padding: 0 30% 0 30%;
```
this can be padding: 0 30%

jennetturkkan commented on src/containers/BlogPage/index.scss:
```diff
@@ -0,0 +1,21 @@
+  font-family: $mainFontFamily;
+
+  .ourBlogTitle {
+    size: 1.75rem;
```
is it a font-size?

louisrli commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,43 @@
+import { Container } from "react-bootstrap";
+import { useTranslation } from "react-i18next";
+import "./index.scss";
+
```
```
const POSTS_PER_PAGE = 9
const BLOG_API_URL = `blahblah?per_page=${POSTS_PER_PAGE}`
```

louisrli commented on src/containers/BlogPage/index.scss:
```diff
@@ -0,0 +1,21 @@
+    padding: 0 30% 0 30%;
+  }
+
+  //Bootstrap class for search bar part
```
space before B

mohammad-jundi committed:
fixed bugs

mohammad-jundi committed:
test run

mohammad-jundi committed:
updated with div, conflict fixed

mohammad-jundi commented:
PTAL @louisrli

louisrli commented on src/components/Blogs/__snapshots__/index.test.jsx.snap:
```diff
@@ -0,0 +1,7 @@
+
+exports[`Blogs renders correctly and matches the snapshot 1`] = `
+<div
+  className="container"
```
this test should show at least one mock blog post. you will have to mock fetch so that it returns some blog post data. google "mock fetch jest unit test" or work with ammar in this one. but it will be worth your time to do this, testing is by far the most popular interview question I've seen from companies so consider yourself lucky to get the chance to do this one😀

mohammad-jundi replied:
I always thought I am the lucky one, now I am sure that I am :)

louisrli commented on src/components/Blogs/index.jsx:
```diff
@@ -0,0 +1,44 @@
+            <h3
+              className="blogPostTitle"
+              dangerouslySetInnerHTML={{
+                __html: t(blogPost.title.rendered),
```
hmm I'm confused. this isn't coming from the translations file. work with hiba or jennet here to solve this, I'm not sure what this page is supposed to be so I can't advise

mohammad-jundi replied:
You are right this is not from the translations file, this one is being fetched from WP API to show only the titles for the posts. I had to use `dangerouslySetInnerHTML` because the data was fetched with extra `&nbsp;` that wasn't supposed to be shown.

louisrli replied:
right but it has a t() around it -- i think it's not supposed to

mohammad-jundi committed:
removed translation in Blogs

mohammad-jundi committed:
search function added

mohammad-jundi committed:
updated with dev

mohammad-jundi committed:
added search function in blogs

mohammad-jundi commented:
@louisrli 
Added the search function, is it ok if the test updated after merging,? if you don't have any other comments of course

louisrli commented:
> @louisrli
> Added the search function, is it ok if the test updated after merging,? if you don't have any other comments of course

no, because i already know that means the test will not get added. and if this were work at a company, you'd have to figure it out anyway :)

mohammad-jundi committed:
adding test to blogs

mohammad-jundi committed:
test run

mohammad-jundi committed:
updated with dev

mohammad-jundi committed:
bugs fixed

mohammad-jundi commented:
@louisrli PTAL

louisrli reviewed (changes requested):
why are 13000 lines changed

mohammad-jundi commented:
> why are 13000 lines changed

sometimes vsc says the files are changing but actually they are not, and the this problem solution is to commit them even though there is no change at all. This time changed 13 000 line !!! I will try to go back to the previous commits and fix. 

update: 
after checking the files, it seems that yarn.lock file have double files in it (I don't know how or why). I am working on resolving it.

hiba-machfej committed:
removed scss and added row

hiba-machfej committed:
removed commented out code

hiba-machfej committed:
Merge branch 'dev' into blogs-page

hiba-machfej commented:
I noticed when opening the branch that carousels on the home page were not there anymore. I noticed that global classes were used like .container or .col which was affecting the home page. I just removed the scss related to container and col and added Row from bootstrap and styling of the home page is working now.  this is how it was appearing before :
![Screen Shot 2020-11-19 at 1 39 40 PM](https://user-images.githubusercontent.com/58190902/99655475-d4cd7000-2a6c-11eb-81b8-3bb69606e532.png)
![Screen Shot 2020-11-19 at 1 39 53 PM](https://user-images.githubusercontent.com/58190902/99655599-f75f8900-2a6c-11eb-92fc-0240f5a4e968.png)

louisrli commented on public/locales/en/translation.json:
```diff
@@ -256,6 +256,9 @@
   },
+  "blogPage": {
+    "title": "Our Blog",
+    "subtitle": "Information about violence, women and health. Read different topics that contribute positively to the mental health of women."},
```
} on new line

louisrli commented on src/components/Blogs/__snapshots__/index.test.jsx.snap:
```diff
@@ -0,0 +1,40 @@
+</div>
+`;
+
+exports[`Blogs renders title: Women standing together 1`] = `undefined`;
```
this does not seem right, it is undefined

louisrli commented on src/components/Blogs/index.test.jsx:
```diff
@@ -0,0 +1,27 @@
+  });
+
+  it("renders title", async () => {
+    Blogs.fetchBlogPosts = jest.fn();
```
this is what your test is doing

1. create a random new function called Blogs.fetchBlogPosts that does nothing (this is not the same as the function in your thing)
2. call this function. console.log title here. it's undefined
3. compare this "undefined" title to a snapshot that also says "undefined."

louisrli commented on src/components/Blogs/index.test.jsx:
```diff
@@ -0,0 +1,27 @@
+
+describe("Blogs", () => {
+  it("renders correctly and matches the snapshot", () => {
+    const tree = renderer.create(<Blogs />).toJSON();
```
```
   const fetchedBlogPosts = await data.json();
    setBlogPosts(fetchedBlogPosts);
```

see this line in your code

you want what you return on line 9-11 to be valid data so that it makes your snapshot test show a blog

please ask for another review when the snapshot test shows an example of a blog post (i can tell that you're not even checking your snapshot test because of what it says right now, otherwise you'd know that the test you wrote doesn't work)

the idea here is that you are mocking fetch so that it gives you fake wordpress data which then creates an example of a blog post in the snapshot test

work with your teammates or ammar to get this finished today please. @jennethydyrova @Jamil-Bailony @hiba-machfej @Ammar-64

mohammad-jundi committed:
updated with dev

mohammad-jundi committed:
conflict solving

mohammad-jundi commented:
> why are 13000 lines changed

I was checking the files, regarding the changed lines. they are from the packages that installed recently, when running yarn install to update the branch with dev it deleted the old package and downloaded the new one which caused the changes in 13000 lines. I am afraid if I remove the file and download it again it will still give me 13000. please advise if it's normal thing to have @louisrli

mohammad-jundi commented:
> I noticed when opening the branch that carousels on the home page were not there anymore. I noticed that global classes were used like .container or .col which was affecting the home page. I just removed the scss related to container and col and added Row from bootstrap and styling of the home page is working now. this is how it was appearing before :
> ![Screen Shot 2020-11-19 at 1 39 40 PM](https://user-images.githubusercontent.com/58190902/99655475-d4cd7000-2a6c-11eb-81b8-3bb69606e532.png)
> ![Screen Shot 2020-11-19 at 1 39 53 PM](https://user-images.githubusercontent.com/58190902/99655599-f75f8900-2a6c-11eb-92fc-0240f5a4e968.png)

Thanks it looks much more better now

mohammad-jundi committed:
fixed typo in translation.json

jennetturkkan committed:
add snapshot test

mohammad-jundi committed:
removed yarn.lock

mohammad-jundi committed:
added yarn.lock

jennetturkkan committed:
work on test

jennetturkkan committed:
try fix test

jennetturkkan committed:
update test

jennetturkkan committed:
fix test

jennetturkkan committed:
Merge branch 'blogs-page' of github.com:ReCoded-Org/capstone-istanbul-safe_place into blogs-page

jennetturkkan committed:
update snapshot

jennetturkkan commented:
@louisrli I wrote the test, but I am not sure if it want you wanted. can you please check?

jennetturkkan committed:
remove unused code

louisrli commented on src/components/Blogs/index.test.jsx:
```diff
@@ -0,0 +1,36 @@
+  });
+
+  it("fetches and renders mock data correctly", async () => {
+    await act(async () => render(<Blogs />));
```
it seems like it works but you only need one test, and that's the snapshot test, and so this rendering should be moved into the snapshot test

also see: 
![Uploading image.png…]()

louisrli commented on src/components/Blogs/index.test.jsx:
```diff
@@ -0,0 +1,36 @@
+  });
+
+  it("fetches and renders mock data correctly", async () => {
+    await act(async () => render(<Blogs />));
```
oops wrote this twice

it seems like it works but you only need one test, and that's the snapshot test, so delete this one and render the snapshot so that the snapshot test shows a blog post please

https://reactjs.org/docs/testing-recipes.html#data-fetching

jennetturkkan replied:
I moved test. can you check?

louisrli replied:
but what about the updated snapshot

jennetturkkan replied:
blogs are not being shown in snapshot, because I think useEffect doesn't run in snapshot

louisrli replied:
well, you have two calls to `<blogs>` right now. perhaps try having one.

jennetturkkan committed:
move test

louisrli commented:
the images seem squahsed in the screenshot jundi put too

jennetturkkan commented:
> the images seem squahsed in the screenshot jundi put too

styling will be updated

louisrli reviewed (commented):
i suggest reading the doc i linked and thinking how it can be applied to this code

louisrli commented:
the only difference should be what you assert at the end of the test (they do some "is it in the document stuff", you would do a snapshot)

jennetturkkan committed:
fix test

jennetturkkan commented:
@louisrli test works. can you check?

louisrli commented on src/components/Blogs/index.test.jsx:
```diff
@@ -0,0 +1,45 @@
+
+let container;
+
+beforeEach(() => {
```
what happens if you delete lines 10 through 18? and line 8

jennetturkkan replied:
if I delete, what can I compare snapshot with?

louisrli commented on src/components/Blogs/index.test.jsx:
```diff
@@ -0,0 +1,45 @@
+    );
+
+    await act(async () => {
+      ReactDOM.render(<Blogs />, container);
```
and just use render(<Blogs/>)

jennetturkkan replied:
it doesn't work without ReactDOM.render(). I tried with render()

louisrli replied:
sry maybe not render. i don't remember but just follow the normal format for snapshots.

```
const tree = renderer.create(<MemberCard />).toJSON();
  expect(tree).toMatchSnapshot();
```

jennetturkkan replied:
it turns out I was importing the wrong render. Now I changed it and it works.

louisrli commented on src/components/Blogs/index.test.jsx:
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import ReactDOM from "react-dom";
```
a bunch of these imports are unused too i believe

jennetturkkan replied:
yes, i will delete them now

jennetturkkan committed:
delete unused code, add render

jennetturkkan committed:
add row, fix img styling

jennetturkkan committed:
fix header style

jennetturkkan committed:
add margin to row, update snapshot test

jennetturkkan commented:
Fixed images and header styling, added some margin, added row

<img width="1440" alt="Screen Shot 2020-11-20 at 14 50 18" src="https://user-images.githubusercontent.com/67760504/99797194-d5830680-2b3f-11eb-98f5-9d085a161442.png">

jennetturkkan committed:
delet unused code

jennetturkkan committed:
nest styling

jennetturkkan committed:
Merge branch 'dev' into blogs-page

louisrli commented on src/components/Blogs/index.test.jsx:
```diff
@@ -0,0 +1,43 @@
+    );
+
+    await act(async () => {
+      render(<Blogs />, container);
```
the thing is, the example code i showed you doesn't use render. it uses render**er**
```
const tree = renderer.create(<MemberCard />).toJSON();
  expect(tree).toMatchSnapshot();
```

basically, i'm fairly sure you don't need to do this whole container thing. just literally copy the code above and put it in the async, change it to Blogs.

jennetturkkan replied:
like this? 
```describe("Blogs", () => {
  it("Renders without crashing", async () => {
    const mockData = [
      {
        title: {
          rendered: "You can find safe place for yourself here",
        },
        jetpack_featured_media_url:
          "https://safeplace102505649.files.wordpress.com/2020/11/womentogether.jpg",
        id: 78,
      },
    ];

    jest.spyOn(global, "fetch").mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockData),
      })
    );

    await act(async () => {
      const tree = renderer.create(<Blogs />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
```

louisrli replied:
```
      expect(tree).toMatchSnapshot();
```

this needs to be outside the act i believe

you are basically awaiting the fetch using act

and THEN you want to match the snapshot

jennetturkkan replied:
then tree will be undefined or should i declare it outside of function?

jennetturkkan replied:
I tried this before too. It doesn't work
``` 
let tree;
    await act(async () => {
      tree = renderer.create(<Blogs />).toJSON();
    });
    expect(tree).toMatchSnapshot(); 
```

jennetturkkan replied:
this is what I get
<img width="1440" alt="Screen Shot 2020-11-20 at 16 12 14" src="https://user-images.githubusercontent.com/67760504/99803921-47ad1880-2b4b-11eb-8f75-6031fb8766ca.png">

louisrli commented:
try it, does it work? :)

On Fri, Nov 20, 2020 at 4:01 PM jennethydyrova <notifications@github.com>
wrote:

> *@jennethydyrova* commented on this pull request.
> ------------------------------
>
> In src/components/Blogs/index.test.jsx
> <https://github.com/ReCoded-Org/capstone-istanbul-safe_place/pull/75#discussion_r527675729>
> :
>
> > +          rendered: "You can find safe place for yourself here",
> +        },
> +        jetpack_featured_media_url:
> +          "https://safeplace102505649.files.wordpress.com/2020/11/womentogether.jpg",
> +        id: 78,
> +      },
> +    ];
> +
> +    jest.spyOn(global, "fetch").mockImplementation(() =>
> +      Promise.resolve({
> +        json: () => Promise.resolve(mockData),
> +      })
> +    );
> +
> +    await act(async () => {
> +      render(<Blogs />, container);
>
> like this?
> `describe("Blogs", () => {
> it("Renders without crashing", async () => {
> const mockData = [
> {
> title: {
> rendered: "You can find safe place for yourself here",
> },
> jetpack_featured_media_url:
> "https://safeplace102505649.files.wordpress.com/2020/11/womentogether.jpg
> ",
> id: 78,
> },
> ];
>
> jest.spyOn(global, "fetch").mockImplementation(() =>
>   Promise.resolve({
>     json: () => Promise.resolve(mockData),
>   })
> );
>
> await act(async () => {
>   const tree = renderer.create(<Blogs />).toJSON();
>   expect(tree).toMatchSnapshot();
> });
>
> });
> });
> `
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ReCoded-Org/capstone-istanbul-safe_place/pull/75#discussion_r527675729>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJ6G2P6HQRZRQ3IUG7ZUOTSQZSC7ANCNFSM4TRGKGWA>
> .
>

jennetturkkan commented:
I tried it before and it doesn't. that's why I added this container thing. 
<img width="1440" alt="Screen Shot 2020-11-20 at 16 04 10" src="https://user-images.githubusercontent.com/67760504/99803236-27308e80-2b4a-11eb-9603-4328fad41be7.png">

louisrli reviewed (approved):
ok thats fine then
